app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+package com.linkedin.drelephant.exceptions;
+
+//Given a log string, analyzes and returns the relevant information!
+import org.apache.log4j.Logger;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AnalyzedLog {
+
+        private static final Logger logger = Logger.getLogger(AnalyzedLog.class);
+        private String state;
+        private ExceptionLoggingEvent exception;
+        private Set<String> list;
+        private String log;
+
+
+        public AnalyzedLog(String log){
+                Matcher m;
+                this.state = "";
+                this.exception = new ExceptionLoggingEvent();
+                this.list = new HashSet<String>();
+                Set<String> allMatches = new HashSet<String>();
+                this.log=log;
+
+                if (match("Flow \\'\\' is set to SUCCEEDED in [0-9]+ seconds").find()) {                                //Successful Azkaban flow log
+                        this.state = "SUCCEEDED";
+                }
+
+                else if (match("Setting flow \\'\\' status to FAILED in [0-9]+ seconds").find()) {                      //Failed Azkaban flow log
+                                m = match("Job (.*) finished with status (?:FAILED|KILLED) in [0-9]+ seconds");
+                                while(m.find()){
+                                        allMatches.add(m.group(1));
+                                }
+                                this.state = "FAILED";
+                                this.list = allMatches;
+
+                }
+
+                else if (match("Setting flow \'\' status to KILLED in [0-9]+ seconds").find()){                         //Killed Azkaban flow log
+                                m = match("Job (.*) finished with status (?:FAILED|KILLED) in [0-9]+ seconds");
+                                while(m.find()){
+                                        allMatches.add(m.group(1));
+                                }
+                                this.state = "KILLED";
+                                this.list = allMatches;
+                }
+
+                else if (match("Finishing job [^\\s]+ attempt: [0-9]+ at [0-9]+ with status SUCCEEDED").find()){        //Succeeded Azkaban Job log
+                        this.state = "SUCCEEDED";
+                }
+
+                else if(match("Finishing job [^\\s]+ attempt: [0-9]+ at [0-9]+ with status FAILED").find()){            //Failed Azkaban Job log
+                        m = match("job_[0-9]+_[0-9]+");
+                        while(m.find()){
+                                allMatches.add(m.group());
+                        }
+                        if(match("ERROR - Job run failed!").find()){
+                                this.state = "SCRIPT FAILED";
+                                this.exception.setType("script");
```

Comment:
Similar to comment above. You can have the execption type as enum instead of hardcoding it.